src/runtime/entity.ts

```diff
@@ -57,12 +62,23 @@ export type EntityClass = (new (data, userIDComponent?: string) => EntityInterfa
 export abstract class Entity implements EntityInterface {
   private userIDComponent?: string;
 
+  private schema: Schema;
   protected rawData: EntityRawData;
 
-  protected constructor(userIDComponent?: string) {
+  protected constructor(data: EntityRawData, schema: Schema, context: ParticleExecutionContext, userIDComponent?: string) {
```

Comment:
I'm not sure that it makes sense to have to construct entities around a ParticleExecutionContext; is there any way to factor that out?

Reply:
Discussed offline, we need them to dereference References. Added a TODO.